### PR TITLE
[feat][python] Support Pytorch task in python api

### LINKED ISSUE #11924: [Feature][PyDolphinScheduler] Support Pytorch task in pyds
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

part of https://github.com/apache/dolphinscheduler/issues/11906

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jieguangzhou**
## Purpose of the pull request

close:  #11924

- [x] add pyds Pytorch task
- [x] add YAML example
- [x] add doc

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

### CODE REVIEW

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/pytorch.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    :param name: task name
+    :param script: Entry to the Python script file that you want to run.
+    :param script_params: Input parameters at run time.
+    :param project_path: The path to the project. Default "." .
+    :param is_create_environment: is create environment. Default False.
+    :param python_command: The path to the python command. Default "${PYTHON_HOME}".
+    :param python_env_tool: The python environment tool. Default "conda".
+    :param requirements: The path to the requirements.txt file. Default "requirements.txt".
+    :param conda_python_version: The python version of conda environment. Default "3.7".
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/pytorch.py:47]: Thanks for adding this doc string

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/pytorch.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self,
+        name: str,
+        script: str,
+        script_params: str = "",
+        project_path: str = DEFAULT.project_path,
+        is_create_environment: bool = DEFAULT.is_create_environment,
+        python_command: str = DEFAULT.python_command,
+        python_env_tool: str = "conda",
+        requirements: str = "requirements.txt",
+        conda_python_version="3.7",
```
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/pytorch.py]: If we have a default value of the parameter, we should better a `Optional` keyword

```suggestion
        is_create_environment: Optional[bool] = DEFAULT.is_create_environment,
        python_command: Optional[str] = DEFAULT.python_command,
        python_env_tool: Optional[str] = "conda",
        requirements: Optional[str] = "requirements.txt",
        conda_python_version: Optional[str] ="3.7",
```

**Comment by jieguangzhou** [comment]
@zhongjiajie PTAL, thanks

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks

**Pull request merged**